operation/manager.go

```diff
@@ -99,6 +111,8 @@ func (sm *SingleOperationManager) WaitTillNotPowered(ctx context.Context, pollTi
 			oldOp := sm.currentOp == ctx.Value(somCtxKeySingleOp)
 			sm.mu.Unlock()
 
+			// Dan: Now that cancelation blocks, I think this `if` statement is a tautology and
```

Comment:
Options for this:
* Agree this is true and I'll make the change now to always call `stop`
* Demonstrate the claim is incorrect and I'll remove the comment
* Remain unconvinced one way or the other and I'll leave the comment in place and make no further changes.

Reply:
So this function is now always called a stop on a powered interface upon a context cancellation, which we're okay with because cancellation ensures that an operation is canceled.

I'm not sure this function makes a lot of long term sense, I think it's a weirdly done convenience function and everything you need to block in a driver is actually in `WaitForSuccess`. So, I'm happy with making this stop upon already achieved cancellation. Let me know if you see any weird behaviour in encoded motor where this thing is used.

Reply:
What if in between the time the lock is unlocked and we check sm.currentOp, that a new op has been created?

Reply:
> What if in between the time the lock is unlocked and we check sm.currentOp, that a new op has been created?

Good question -- a new operation cannot be created because new operations [will call the `cancel[AndWait]Func` if they see `sm.currenOp != nil`](https://github.com/viamrobotics/rdk/pull/2667/files#diff-6163f94661630679b64f82eba6259cc10b6cbec32a16fa47326e87a5f61d6b8cR55-R57).

So I think the (simplified) [code example](https://github.com/viamrobotics/rdk/blob/84bd37abd428b9d7a5b429ebe49f9a0093c18bf7/components/motor/gpiostepper/gpiostepper.go#L295-L319) that shows how this all works together:
```
func (c *Component) MyMovement(ctx, instructions) {
  ctx, done := c.som.New(ctx) // <-- registers som.currentOp to a non-nil
  defer done() // <-- resets som.currentOp back to nil
  c.powerUp(instructions)

  return sm.WaitTillNotPowered(ctx, Second, c, c.Stop)
  // execute `done()` -- even though `sm.mu` is unlocked, the single op manager is providing synchronization due to the existence of a non-nil `som.currentOp`. No other goroutines can register an op while this thread is inside `WaitTillNotPowered`.
}
```

So if that's convincing (I admittedly do not feel 100% about the argument -- too many moving pieces). I think the question becomes -- do we need to take the mutex on the above lines [just to read](https://github.com/viamrobotics/rdk/pull/2667/files#diff-6163f94661630679b64f82eba6259cc10b6cbec32a16fa47326e87a5f61d6b8cR110-R112) `som.currentOp`? I think the answer is we don't. But I also don't feel good having my name show up there as well in a `git blame`!

> So, I'm happy with making this stop upon already achieved cancellation. Let me know if you see any weird behaviour in encoded motor where this thing is used.

Sounds like a vote for Option (1). Assuming you've grokked my argument @edaniels, any strong opinion here?

Reply:
agree with option 1

Reply:
Option (1) pushed